app/src/main/java/com/expidev/gcmapp/Constants.java

```diff
@@ -4,6 +4,7 @@
     /* SharedPreference constants */
     public static final String PREFS_SETTINGS = "gcm_prefs";
     public static final String PREF_CURRENT_MINISTRY = "currentMinistry";
+    public static final String PREF_CURRENT_ASSIGNMENT = "currentAssignment"; // stores the assignment ID
```

Comment:
I just added a comment with a potential workaround the main place you were utilizing this preference value